File: backend/src/services/storage.service.ts
import * as Minio from 'minio';

const minioClient = new Minio.Client({
  endPoint: process.env.MINIO_ENDPOINT || 'localhost',
  port: parseInt(process.env.MINIO_PORT || '9000'),
  useSSL: false,
  accessKey: process.env.MINIO_ACCESS_KEY || 'minioadmin',
  secretKey: process.env.MINIO_SECRET_KEY || 'minioadmin'
});

const bucketName = process.env.MINIO_BUCKET || 'documents';

export const initializeStorage = async () => {
  try {
    const exists = await minioClient.bucketExists(bucketName);
    if (!exists) {
      await minioClient.makeBucket(bucketName, 'us-east-1');
      console.log(`Created MinIO bucket: ${bucketName}`);
    }
  } catch (err) {
    console.error('MinIO initialization error:', err);
  }
};

export const uploadFile = async (objectName: string, buffer: Buffer, mimetype: string) => {
  const metaData = {
    'Content-Type': mimetype,
  };
  await minioClient.putObject(bucketName, objectName, buffer, buffer.length, metaData);
};

export const getFileUrl = async (objectName: string) => {
  return minioClient.presignedGetObject(bucketName, objectName, 24 * 60 * 60); // 24 hours
};

export const deleteFile = async (objectName: string) => {
  await minioClient.removeObject(bucketName, objectName);
};

File: backend/app/services/storage_service.py
import os

class StorageService:
    def __init__(self):
        self.endpoint = os.environ.get("MINIO_ENDPOINT")
        self.access_key = os.environ.get("MINIO_ACCESS_KEY")
        self.secret_key = os.environ.get("MINIO_SECRET_KEY")
        self.bucket = os.environ.get("MINIO_BUCKET")
        # Initialize boto3 or minio client here when implemented

    def upload(self, file_path: str, object_name: str):
        pass

    def download(self, object_name: str, file_path: str):
        pass

    def delete(self, object_name: str):
        pass

    def exists(self, object_name: str) -> bool:
        return False

    def generate_presigned_url(self, object_name: str) -> str:
        return ""
